src/modules/sensors.cpp

```diff
@@ -18,22 +20,30 @@ static struct pt sensorsThread;
 static struct pt sensorsPressureThread;
 static struct pt sensorsAirFlowThread;
 static struct pt sensorsBatteryThread;
+static struct timer pressureTimer;
+static struct timer airflowTimer;
 
 uint32_t last_debug_time;
 
+#define PRESSURE_SAMPLING_RATE 50000
 static PT_THREAD(sensorsPressureThreadMain(struct pt* pt))
 {
   PT_BEGIN(pt);
+  timerHalBegin(&pressureTimer, PRESSURE_SAMPLING_RATE);  // set to 50ms to not overflow timer. Can be slowed down once timer hal is updated.
+  PT_WAIT_UNTIL(pt, timerHalRun(&pressureTimer)!=HAL_IN_PROGRESS);
```

Comment:
If we put this right before `PT_RESTART` any time taken to get the sample and do the processing will be included in the timeout, making the sample rate more consistent (though barely, I dont expect it to take more than 1ms to do everything, but maybe it might take 2ms in the worst case if our math for derived information gets fancy)

Reply:
Did you mean to move the PT_RESTART up to right beneath the PT_WAIT_UNTIL? I tried that, and it mucked things up with the debug reporting. Maybe we can sync and get the debug stuff working properly, and then this should work fine?

Reply:
Fixed after Mike set me straight. Done.